platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/merkle/AbstractHashListener.java

```diff
@@ -121,70 +115,41 @@ public synchronized void onHashingStarted() {
     @Override
     public void onNodeHashed(final long path, final Hash hash) {
         assert hashes != null && leaves != null : "onNodeHashed called without onHashingStarted";
-        final List<VirtualHashRecord> dirtyHashesToFlush;
-        final List<VirtualLeafRecord<K, V>> dirtyLeavesToFlush;
-        synchronized (this) {
-            hashes.add(new VirtualHashRecord(path, hash));
-            if ((reconnectFlushInterval > 0)
-                    && (hashes.size() >= reconnectFlushInterval)
-                    && flushInProgress.compareAndSet(false, true)) {
-                dirtyHashesToFlush = hashes;
-                hashes = new ArrayList<>();
-                dirtyLeavesToFlush = leaves;
-                leaves = new ArrayList<>();
-            } else {
-                dirtyHashesToFlush = null;
-                dirtyLeavesToFlush = null;
-            }
-        }
-        if ((dirtyHashesToFlush != null) && (dirtyLeavesToFlush != null)) {
-            flush(dirtyHashesToFlush, dirtyLeavesToFlush);
+        hashes.add(new VirtualHashRecord(path, hash));
```

Comment:
Two concerns:

1. Here in this class (which is used during reconnects): synchronization in hash listener methods is not needed any more as notifications are now serialized. Fine. However, when a flush happens, it would block the current notification task and all subsequent tasks until the flush is complete. Previously, it would block only one task, but other tasks could accumulate hashes/leaves in parallel

2. During regular hashing in the end of each round, when hashes are put to `VirtualNodeCache`, it was done in parallel (and the cache is ready to handle that), but with this new approach it will be done sequentially. I wonder how copy hash time is affected by this change?